### PR TITLE
Emmyoop patch 3

### PR DESCRIPTION
**Pull request by emmyoop**
## Description & motivation
Updating docs for some dbt-core v1.0.0 changes.

https://github.com/dbt-labs/dbt/pull/4008
https://github.com/dbt-labs/dbt/pull/4007
https://github.com/dbt-labs/dbt/pull/3999

## To-do before merge
dbt-core v1.0.0 needs to be released

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [x] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [ ] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

## Checklist
Renamed pages:
- [x] The pages have been updated in `website/sidebars.js`
- [x] The pages have unique filenames
- [x] Entries have been added to `_redirects`

### CODE REVIEW

**Comment by emmyoop** [comment]
Unsure of where to put changelog entries for each change.

File: website/docs/docs/building-a-dbt-project/exposures.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ Exposures make it possible to define and describe a downstream use of your dbt p
 
 ### Declaring an exposure
 
-Exposures are defined in `.yml` files in your `models` directory (as defined by the [`source-paths` config](source-paths)), nested under an `exposures:` key.
+Exposures are defined in `.yml` files in your `models` directory (as defined by the [`model-paths` config](model-paths)), nested under an `exposures:` key.
```
Comment (jtcohen6) [inline website/docs/docs/building-a-dbt-project/exposures.md]: Exposures can actually be defined in any resource path (`models/`, `seeds/`, `macros/`, `tests/`, `analyses/`). So this statement isn't exactly wrong, but it does play into just a common misconception

File: website/docs/docs/building-a-dbt-project/using-sources.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ Sources make it possible to name and describe the data loaded into your warehous
 
 ### Declaring a source
 
-Sources are defined in `.yml` files in your `models` directory (as defined by the [`source-paths` config](source-paths)), nested under a `sources:` key.
+Sources are defined in `.yml` files in your `models` directory (as defined by the [`model-paths` config](model-paths)), nested under a `sources:` key.
```
Comment (jtcohen6) [inline website/docs/docs/building-a-dbt-project/using-sources.md]: Same comment as for exposures above

File: website/docs/docs/guides/migration-guide/upgrading-to-1-0-0.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+---
+
+### Resources
+
+- [Discourse]()
+- [Release notes](https://github.com/dbt-labs/dbt/releases/tag/v1.0.0)
+- [Full changelog](https://github.com/dbt-labs/dbt/blob/1.0.0/CHANGELOG.md)
+
+## Breaking changes
+- Do I need to mention enacted deprecations?
```
Comment (jtcohen6) [inline website/docs/docs/guides/migration-guide/upgrading-to-1-0-0.md]: Yes please!

File: website/docs/faqs/configurable-data-path.md [hunk 4]
```diff
@@ -1,16 +1,16 @@
 ---
-title: Can I store my seeds in a directory other than the `data` directory in my project?
+title: Can I store my seeds in a directory other than the `seeds` directory in my project?
```
Comment (jtcohen6) [inline website/docs/faqs/configurable-data-path.md:2]: I definitely think a `<Changelog>` entry here makes sense

File: website/docs/faqs/configurable-data-test-path.md [hunk 5]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Can I store my data tests in a directory other than the `test` directory in my project?
+title: Can I store my data tests in a directory other than the `tests` directory in my project?
```
Comment (jtcohen6) [inline website/docs/faqs/configurable-data-test-path.md:2]: by contrast, I don't really feel a need for a `<Changelog>` entry with this one :)

File: website/docs/faqs/configurable-model-path.md [hunk 6]
```diff
@@ -3,13 +3,13 @@ title: Can I store my models in a directory other than the `models` directory in
 ---
 By default, dbt expects your seed files to be located in the `models` subdirectory of your project.
 
-To change this, update the [source-paths](reference/project-configs/source-paths.md) configuration in your `dbt_project.yml`
+To change this, update the [model-paths](reference/project-configs/model-paths.md) configuration in your `dbt_project.yml`
```
Comment (jtcohen6) [inline website/docs/faqs/configurable-model-path.md:8]: Changelog entry

File: website/docs/reference/analysis-properties.md [hunk 7]
```diff
@@ -4,7 +4,7 @@ title: Analysis properties
 
 Analysis properties can be declared in `.yml` files in:
 - your `analyses/` directory (as defined by the [`analysis-paths` config](analysis-paths))
-- your `models/` directory (as defined by the [`source-paths` config](source-paths))
+- your `models/` directory (as defined by the [`model-paths` config](model-paths))
```
Comment (jtcohen6) [inline website/docs/reference/analysis-properties.md]: As with sources and exposures — analysis properties (like all properties) can actually be defined in _any_ `.yml` file, in _any_ resource path.

Maybe we could rephrase this to say: We highly encourage you to define analysis properties in your `analyses/` directory?

(hilarious that this already said `analyses/` and not `analysis`/)

File: website/docs/reference/commands/clean.md [hunk 8]
```diff
@@ -3,4 +3,4 @@ title: "clean"
 id: "clean"
 ---
 
-`dbt clean` is a utility function that deletes all folders specified in the `clean-targets` list specified in `dbt_project.yml`. This is useful for deleting the `dbt_modules` and `target` directories.
\ No newline at end of file
+`dbt clean` is a utility function that deletes all folders specified in the `clean-targets` list specified in `dbt_project.yml`. This is useful for deleting the `dbt_packages` and `target` directories.
```
Comment (jtcohen6) [inline website/docs/reference/commands/clean.md:12]: Worth a `<Changelog>` here, since this will require action from folks to manually delete `dbt_modules`

File: website/docs/reference/commands/seed.md [hunk 9]
```diff
@@ -3,7 +3,7 @@ title: "seed"
 id: "seed"
 ---
 
-The `dbt seed` command will load `csv` files located in the `data-paths` directory of your dbt project into your data warehouse.
+The `dbt seed` command will load `csv` files located in the `seed-paths` directory of your dbt project into your data warehouse.
```
Comment (jtcohen6) [inline website/docs/reference/commands/seed.md:13]: changelog - this feels important

File: website/docs/reference/configs-and-properties.md [hunk 10]
```diff
@@ -52,10 +52,10 @@ In dbt, properties are declared in `.yml` files, in the same directory as your r
...
-| sources   | `models/`                 | [source-paths](source-paths) |
-| exposures | `models/`                 | [source-paths](source-paths) |
-| seeds     | `data/` or `models/`      | [data-paths](data-paths)     |
+| models    | `models/`                 | [model-paths](model-paths) |
+| sources   | `models/`                 | [model-paths](model-paths) |
+| exposures | `models/`                 | [model-paths](model-paths) |
+| seeds     | `seeds/` or `models/`      | [seed-paths](seed-paths)     |
 | snapshots | `snapshots/` or `models/` | [snapshot-paths](snapshot-paths) |
 | analyses  | `analyses/` or `models/`  | [analysis-paths](analysis-paths) |
 | macros    | `macros/` or `models/`    | [macro-paths](macro-paths) |
```
Comment (jtcohen6) [inline website/docs/reference/configs-and-properties.md]: hmm - can we update these?
- sources and exposures can be defined in any resource path
- I don't think `models/` will keep working for some of these (and it was there for backwards compatibility)

File: website/docs/reference/dbt_project.yml.md [hunk 11]
```diff
@@ -16,8 +16,8 @@ dbt uses YAML in a few different places. If you're new to YAML, it would be wort
 
 [profile](project-configs/profile): profilename
 
-[source-paths](project-configs/source-paths): [directorypath]
-[data-paths](project-configs/data-paths): [directorypath]
+[model-paths](project-configs/model-paths): [directorypath]
```
Comment (jtcohen6) [inline website/docs/reference/dbt_project.yml.md:25]: Definitely a `<Changelog>` for this page

File: website/docs/reference/exposure-properties.md [hunk 12]
```diff
@@ -13,7 +13,7 @@ title: Exposure properties
 - [Declaring resource properties](configs-and-properties)
 
 ## Overview
-Exposures are defined in `.yml` files in your `models` directory (as defined by the [`source-paths` config](source-paths)), nested under an `exposures:` key. You may define `exposures` in YAML files that also define define `sources` or `models`.
+Exposures are defined in `.yml` files in your `models` directory (as defined by the [`model-paths` config](model-paths)), nested under an `exposures:` key. You may define `exposures` in YAML files that also define define `sources` or `models`.
```
Comment (jtcohen6) [inline website/docs/reference/exposure-properties.md]: as above/elsewhere

File: website/docs/reference/macro-properties.md [hunk 13]
```diff
@@ -4,7 +4,7 @@ title: Macro properties
 
 Macro properties can be declared in `.yml` files in:
 - your `macros/` directory (as defined by the [`macro-paths` config](macro-paths))
-- your `models/` directory (as defined by the [`source-paths` config](source-paths))
+- your `models/` directory (as defined by the [`model-paths` config](model-paths))
```
Comment (jtcohen6) [inline website/docs/reference/macro-properties.md]: as above/elsewhere, this isn't wrong exactly but it does play into a common misconception

File: website/docs/reference/project-configs/clean-targets.md [hunk 14]
```diff
@@ -23,31 +23,31 @@ If this configuration is not included in your `dbt_project.yml` file, the `clean
 :::info
 This is our preferred configuration
 :::
-To remove packages as well as compiled files, include the value of your [modules-path](modules-path) configuration in your `clean-targets` configuration.
+To remove packages as well as compiled files, include the value of your [packages-install-path](packages-install-path) configuration in your `clean-targets` configuration.
```
Comment (jtcohen6) [inline website/docs/reference/project-configs/clean-targets.md:32]: I think a `<Changelog>` here would make a lot of sense, since it will require active change from users when they upgrade to v1

File: website/docs/reference/project-configs/model-paths.md [hunk 15]
```diff
@@ -5,7 +5,7 @@ default_value: [models]
 <File name='dbt_project.yml'>
 
 ```yml
-source-paths: [directorypath]
+model-paths: [directorypath]
```
Comment (jtcohen6) [inline website/docs/reference/project-configs/model-paths.md:14]: definitely `<Changelog>` here

File: website/docs/reference/project-configs/packages-install-path.md [hunk 16]
```diff
@@ -1,11 +1,11 @@
 ---
 datatype: directorypath
-default_value: modules
+default_value: dbt_packages
```
Comment (jtcohen6) [inline website/docs/reference/project-configs/packages-install-path.md:3]: definitely `<Changelog>` here

File: website/docs/reference/project-configs/seed-paths.md [hunk 17]
```diff
@@ -5,7 +5,7 @@ default_value: [data]
 <File name='dbt_project.yml'>
 
 ```yml
-data-paths: [directorypath]
+seed-paths: [directorypath]
```
Comment (jtcohen6) [inline website/docs/reference/project-configs/seed-paths.md:15]: definitely `<Changelog>` here

File: website/docs/reference/seed-properties.md [hunk 18]
```diff
@@ -7,12 +7,12 @@ title: Seed properties
 </Changelog>
 
 Seed properties can be declared in `.yml` files in:
-- your `data/` directory (as defined by the [`data-paths` config](data-paths))
-- your `models/` directory (as defined by the [`source-paths` config](source-paths)), for backwards compatibility
+- your `seeds/` directory (as defined by the [`seed-paths` config](seed-paths))
```
Comment (jtcohen6) [inline website/docs/reference/seed-properties.md]: - a changelog entry for this one feels justified
- [as above/elsewhere] these properties can really be defined anywhere, we should recommend defining them in the `seeds/` directory (as we do below)

File: website/docs/reference/source-properties.md [hunk 19]
```diff
@@ -7,7 +7,7 @@ title: Source properties
 - [Declaring resource properties](configs-and-properties)
 
 ## Overview
-Source properties can be declared in `.yml` files in your `models/` directory (as defined by the [`source-paths` config](source-paths)).
+Source properties can be declared in `.yml` files in your `models/` directory (as defined by the [`model-paths` config](model-paths)).
```
Comment (jtcohen6) [inline website/docs/reference/source-properties.md:10]: or anywhere else!

File: website/docs/tutorial/using-jinja.md [hunk 20]
```diff
@@ -5,7 +5,7 @@ id: "using-jinja"
 
 In this tutorial, we're going to take a common pattern used in SQL, and then use Jinja to improve our code.
 
-If you'd like to work through this query, add [this CSV](https://github.com/dbt-labs/jaffle_shop/blob/master/data/raw_payments.csv) to the `data/` folder of your dbt project, and then execute `dbt seed`.
+If you'd like to work through this query, add [this CSV](https://github.com/dbt-labs/jaffle_shop/blob/HEAD/seeds/raw_payments.csv) to the `seeds/` folder of your dbt project, and then execute `dbt seed`.
```
Comment (jtcohen6) [inline website/docs/tutorial/using-jinja.md]: this will be a broken link until we update the `jaffle_shop` repo - what do you think of leaving this as is for now, and leaving ourselves a `TODO`?

File: website/docs/docs/building-a-dbt-project/package-management.md [hunk 21]
```diff
@@ -42,7 +42,7 @@ packages:
 
 </File>
 
-3. Run `dbt deps` to install the package(s). Packages get installed in the `dbt_modules` directory – by default this directory is ignored by git, to avoid duplicating the source code for the package.
+3. Run `dbt deps` to install the package(s). Packages get installed in the `dbt_packages` directory – by default this directory is ignored by git, to avoid duplicating the source code for the package.
```
Comment (jtcohen6) [inline website/docs/docs/building-a-dbt-project/package-management.md:51]: Hmm maybe `<Changelog>` here?

**Review by jtcohen6 — COMMENTED** [review]
Thank you for these updates !!

I left way too many comments. As a general rule:
- Wherever we're documenting The Thing (`model-paths`, `seed-properties`, etc), we should include a `<Changelog>` item, just calling out that the config/default has been renamed
- If we're just updating the name of a file in an example, no big deal, I don't think we need a changelog
- In a _bunch_ of places, the current docs refer to a limited number of places where sources/exposures/properties can be defined. Instead, we should clarify that they can be defined in _any_ resource path, and (in the case of properties) that we highly encourage defining them in the resource path associated with the resource type they're describing. I think it's confusing to call out `model-paths` as an alternative / backwards-compatible option

As soon as you're happy with these docs, we can merge them into the `next` branch and deploy them at next.docs.getdbt.com. I'm probably going to merge #836 first, so there will be some conflicting changes to resolve/rebase, namely in the migration guide.

File: website/docs/tutorial/using-jinja.md (hunk 20 above)
Comment (emmyoop) [reply website/docs/tutorial/using-jinja.md]: What about linking to the branch where it does exist?  https://github.com/dbt-labs/jaffle_shop/blob/core-v1.0.0/seeds/raw_payments.csv

File: website/docs/reference/configs-and-properties.md (hunk 10 above)
Comment (emmyoop) [reply website/docs/reference/configs-and-properties.md]: Thoughts on just removing this table and leaving the surrounding text?  Since the resource names now line up with the default directories it seems overly verbose?

```
## Where can I define properties?

In dbt, properties are declared in `.yml` files, in the same directory as your resources.

You can name these files `whatever_you_want.yml` and nest them arbitrarily deeply in subfolders within each directory.

We highly recommend that you define properties in dedicated paths alongside the resources they're describing.```

Comment (jtcohen6) [reply website/docs/reference/configs-and-properties.md]: Fantastic idea!

File: website/docs/tutorial/using-jinja.md (hunk 20 above)
Comment (jtcohen6) [reply website/docs/tutorial/using-jinja.md]: that works!

**Review by jtcohen6 — APPROVED** [review]
Thanks for all the hard work to get these right !!

There might be a few stragglers, which we can catch them over the next several weeks — at this point, it feels more important to get the vast majority of the updates in.
